src/Sulu/Bundle/PageBundle/Resources/js/containers/Form/fields/SegmentSelect.js

```diff
@@ -0,0 +1,43 @@
+// @flow
+import React from 'react';
+import {computed} from 'mobx';
+import {SingleSelect} from 'sulu-admin-bundle/components';
+import type {FieldTypeProps} from 'sulu-admin-bundle/types';
+import {translate} from 'sulu-admin-bundle/utils';
+import webspaceStore from '../../../stores/webspaceStore';
+
+export default class SegmentSelect extends React.Component<FieldTypeProps<string>> {
```

Comment:
This has not been implementd like this on purpose. The thing is that the `page_excerpt` form makes use of this `SegmentSelect`, and this form is also used in the ArticleBundle. And in the case of the ArticleBundle it has to be possible to assign a segment per webspace. Since the same form is used in combination with different entities, this field has to implement this logic. If a webspace is set, it should only show a `SingleSelect`, if no webspace is set, it should show a `SingleSelect` for each available webspace.

This has just not been implemented in this PR, because it is so big already. Therefore this field type has to be changed when the ArticleBundle integrates Segments, but it should not be renamed then.

Reply:
sounds reasonable 🙂